### ISSUE TITLE
Error when running "make -j4"

### ISSUE THREAD

**Issue by Jantalus** [SideChannelMarvels/Tracer#38]
Hi! I was hoping I could use the tool you coded, would be of great help. The thing is I followed each step but I'm getting an error:

```
tg_main.c: In function ‘tg_post_clo_init’:
tg_main.c:472:26: error: passing argument 1 of ‘vgPlain_strncat’ from incompatible pointer type [-Wincompatible-pointer-types]
  472 |             VG_(strncat)(buffer, " ", INFO_BUFFER_SIZE);
      |                          ^~~~~~
      |                          |
      |                          char **
In file included from tg_main.c:26:
../include/pub_tool_libcbase.h:84:44: note: expected ‘HChar *’ {aka ‘char *’} but argument is of type ‘char **’
   84 | extern HChar* VG_(strncat)        ( HChar* dest, const HChar* src, SizeT n );
      |                                     ~~~~~~~^~~~
tg_main.c:473:22: error: passing argument 1 of ‘vgPlain_strncat’ from incompatible pointer type [-Wincompatible-pointer-types]
  473 |         VG_(strncat)(buffer, *(HChar**)VG_(indexXA)(VG_(args_for_client), i), INFO_BUFFER_SIZE);
      |                      ^~~~~~
      |                      |
      |                      char **
../include/pub_tool_libcbase.h:84:44: note: expected ‘HChar *’ {aka ‘char *’} but argument is of type ‘char **’
   84 | extern HChar* VG_(strncat)        ( HChar* dest, const HChar* src, SizeT n );
      |                                     ~~~~~~~^~~~
tg_main.c:475:15: error: assignment to ‘const char *’ from incompatible pointer type ‘char **’ [-Wincompatible-pointer-types]
  475 |     msg.value = buffer;
      |               ^
tg_main.c: In function ‘tg_instrument’:
tg_main.c:532:10: warning: declaration of ‘trace_instr’ shadows a global declaration [-Wshadow]
  532 |     Bool trace_instr = False;
      |          ^~~~~~~~~~~
tg_main.c:69:12: note: shadowed declaration is here
   69 | static int trace_instr = 1;
      |            ^~~~~~~~~~~
tg_main.c: In function ‘tg_fini’:
tg_main.c:694:15: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
  694 |     while((di = VG_(next_DebugInfo)(di)) != NULL)
      |               ^
tg_main.c: In function ‘tg_pre_clo_init’:
tg_main.c:715:36: error: passing argument 2 of ‘vgPlain_basic_tool_funcs’ from incompatible pointer type [-Wincompatible-pointer-types]
  715 |                                    tg_instrument,
      |                                    ^~~~~~~~~~~~~
      |                                    |
      |                                    IRSB * (*)(VgCallbackClosure *, IRSB *, VexGuestLayout *, VexGuestExtents *, VexArchInfo *, IRType,  IRType)
```

My current operating system is 6.9.2-arch1-1

**Comment by doegox** [comment]
Hello, I pushed a new branch valgrind-3.23.0 and fixed the code to remove all warnings of my gcc 13.3
Please test (see README instructions in that branch) and report if this works for you or not.
If you've still issues please post them and tell which gcc version you have.

**Comment by Jantalus** [comment]
Hii!! Thanks for the fast response, I wouldn't think that you would respond.
Could do all the steps from the readme, I only had to rename the "tracergrind-3.2.0" to "tracergrind" so the makefile could find the folder. 
Then the only thing that seems like an error is when running "texttrace", previously I installed capstone from the original capstone repo with the 3.0.5 release, following the COMPILE.TXT file in the capstone codebase.
The error output when using the texttrace tool is like the following:

```
...
Disassembly failure at ExecMsg 2010324!
Disassembly failure at ExecMsg 2010325!
Disassembly failure at ExecMsg 2010328!
Disassembly failure at ExecMsg 2010336!
Disassembly failure at ExecMsg 2010338!
Disassembly failure at ExecMsg 2010341!
Disassembly failure at ExecMsg 2010342!
Disassembly failure at ExecMsg 2010351!
Disassembly failure at ExecMsg 2010352!
Disassembly failure at ExecMsg 2010355!
Disassembly failure at ExecMsg 2010363!
Disassembly failure at ExecMsg 2010365!
Disassembly failure at ExecMsg 2010368!
Disassembly failure at ExecMsg 2010372!
Disassembly failure at ExecMsg 2010375!
Disassembly failure at ExecMsg 2010379!
Disassembly failure at ExecMsg 2010383!
Disassembly failure at ExecMsg 2010392!
Disassembly failure at ExecMsg 2010393!
Disassembly failure at ExecMsg 2010394!
Disassembly failure at ExecMsg 2010409!
Disassembly failure at ExecMsg 2010413!
Disassembly failure at ExecMsg 2010419!
Disassembly failure at ExecMsg 2010420!
Disassembly failure at ExecMsg 2010421!
Disassembly failure at ExecMsg 2010437!
Disassembly failure at ExecMsg 2010442!
...
```
my terminal was filled with these messages. Apart from that all good. Will be the trace be complete with these errors? I would guess not.

Thanks for the help!

**Comment by doegox** [comment]
honestly I've no idea :(
try to compare your trace with the actual binary
I cleaned a bit the branch to fix the "tracergrind-3.2.0" and remove the old patches
